File: PO/Project/C++/src/wolf.cpp
#include "wolf.h"

wolf::wolf(world* world_point, YX position)
:animal(9,5, position, world_point)
{
    avatar = 'W';
}

organizm* wolf::MakeChild(YX new_position)
{
    return (new wolf(world_point, new_position));
}

wolf::~wolf()
{
}

File: PO/Project/JavaProject/src/Animals/Wolf.java
package Animals;

import Abstractions.*;
import com.google.gson.JsonObject;

import java.awt.*;

public class Wolf extends Animal {

    public Wolf(World world, Point position) {
        super(9, 5, position, world);
        this.avatar = 'W';
        this.image_path = "Wolf.png";
        loadImage();
    }

    @Override
    public Organism makeChild(Point newPosition) {
        return new Wolf(world, newPosition);
    }
}
